docs/reference/reana-yaml/index.md

```diff
@@ -1,112 +1,219 @@
+The **workflow** property of `reana.yaml` specifies the computational steps
+that are necessary to take to get the results. REANA supports three different
+workflow specification languages
+([CWL](/running-workflows/supported-systems/cwl),
+[Serial](/running-workflows/supported-systems/serial),
+[Yadage](/running-workflows/supported-systems/yadage)). Each workflow
+specification language expresses the computational steps differently.
 
-The `specification` part consists of a list of `steps`, each of these containing
-a list of `commands` to execute. Each step must have an `environment` specified i.e.
-docker image name.
 
-#### Advanced usage
+The **workflow** property is composed of:
 
-You can specify additional requirements such as [compute backend](../../advanced-usage/compute-backends/),
-[CVMFS](../../advanced-usage/code-repositories/cvmfs), [Kerberos access control](../../advanced-usage/access-control/kerberos#setting-kerberos-requirement).
+| Property      | Type         | Mandatory?                                         | Description                                 |
+| --------      | ----         | ----------                                         | -----------                                 |
+| type          | *string*     | *mandatory*                                        | Specifies workflow language type. Can be `cwl`, `serial`, `yadage`. |
+| file          | *string*     | *mandatory if property `specification` is missing* | For CWL and Yadage workflows, specifies workflow steps in an external file, using their respective workflow definition languages.  |
+| specification | *dictionary* | *mandatory if property `file` is missing*          | For Serial workflows, specifies workflow steps internally in `reana.yaml`, see below.  |
 
+The **workflow.specification** property is used in Serial workflows and is further composed of:
 
-### CWL
+| Property | Type   | Mandatory?  | Description                                        |
+| -------- | ----   | ----------  | -----------                                        |
+| steps    | *list* | *mandatory* | Lists all workflow steps that are to be run sequentially to obtain workflow results. |
 
-To be added
+The **workflow.specification.steps** property describes each individual computational step of the Serial workflow and is further composed of:
 
-### YADAGE
+| Property | Type   | Mandatory?  | Description                                        |
+| -------- | ----   | ----------  | -----------                                        |
+| name    | *string* | *optional* | Provides name of the given workflow step. |
+| environment  | *string*  | *mandatory*  | Specifies runtime environment container image where the given workflow step commands will be run.  |
+| commands  | *list*  | *mandatory*  | Lists all commands to be run in the runtime environment container image when the given workflow step is executed.  |
```

Comment:
Perhaps here we can mention that each command will be run in a separate runtime container? Unless we change it. IIRC this has caused some confusion during the Awesome Workshop.

Reply:
Added a brief note, but we'll still have to explain in detail in the overall Serial documentation.